Dockerfile

```diff
@@ -57,24 +64,33 @@ RUN uv sync && uv sync --extra fa
 
 # Runtime stage
 FROM python:3.11-slim
-WORKDIR /root/prime-rl
 
-RUN apt-get update && apt-get install -y --no-install-recommends --force-yes build-essential wget clang
+# 2. Put your code under /appuser not /opt, then hand ownership to appuser ─
+WORKDIR /appuser/prime-rl
+
+# still need a compiler for wheels that ship only as source
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        build-essential wget clang && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create non-root user in runtime stage
+ARG USER_ID=1000
+ARG GROUP_ID=1000
+RUN groupadd --gid $GROUP_ID appuser && \
+    useradd --uid $USER_ID --gid appuser --create-home --shell /bin/bash appuser
+
+# Copy the virtual-env and fix permissions
+COPY --from=builder /appuser/prime-rl/.venv /appuser/prime-rl/.venv
+RUN ln -sf /usr/local/bin/python /appuser/prime-rl/.venv/bin/python && \
+    chown -R appuser:appuser /appuser
```

Comment:
id prefer this in its own line instead of as && as its kind of unrelated to venv